src/include/Parser.h

```diff
@@ -0,0 +1,45 @@
+#pragma once
+
+#include <Lexer.h>
+#include <ASTree.h>
+#include <exception>
+
+class Parser {
+    std::unique_ptr<ASTree> m_tree;
+    Lexer m_lexer;
+    Token m_currentToken;
```

Comment:
Why do you need to store any AST stuff in the parser as a field?

Reply:
do you mean the tree, or `m_lexer` and `m_currentToken` as well?